flink-table/flink-table-planner/src/main/java/org/apache/flink/table/plan/TableOperationConverter.java

```diff
@@ -264,6 +273,28 @@ public RelNode visitOther(TableOperation other) {
 			throw new TableException("Unknown table operation: " + other);
 		}
 
+		@Override
+		public <U> RelNode visitTableSourceTable(TableSourceTableOperation<U> tableSourceTable) {
+			final Table relTable = new TableSourceTable<>(
+				tableSourceTable.getTableSource(),
+				!tableSourceTable.isBatch(),
+				FlinkStatistic.UNKNOWN());
+
+			CatalogReader catalogReader = (CatalogReader) relBuilder.getRelOptSchema();
+			String refId = Integer.toString(System.identityHashCode(tableSourceTable.getTableSource()));
```

Comment:
Can you add an explanation what we are doing here? Btw should we only use the refId or should we prefix it for readability `unregistered_456789`?